pastas/stats/sgi.py

```diff
@@ -22,20 +26,30 @@ def sgi(series: Series) -> Series:
     Notes
     -----
     The Standardized Groundwater Index (SGI) is a non-parametric method to
-    standardize groundwater levels. The SGI is calculated for each month
-    separately. The SGI is a dimensionless index and is used to compare
-    groundwater levels across different wells. It is generally recommended to resample
-    the data to a monthly time series before computing the SGI.
-
+    standardize groundwater levels. The SGI is calculated for each aggregation
+    period within the year separately. The data within that period in all years
+    in the series are used to determine the reference for which the index is
+    calculated for each value in that period.
+    The SGI is a dimensionless index and is used to compare groundwater levels
+    across different wells. It may be useful to resample the time series to a
+    monthly interval before computing the SGI.
     """
+    if timescaleMonths not in (1, 2, 3):
+        raise ValueError(
+            "SGI can only be called with timescaleMonths = 1, 2, or 3; not"
+            + str(timescaleMonths)
+        )
     if isinstance(series, DataFrame):
-        series = series.apply(sgi)
+        series = series.apply(sgi, timescaleMonths=timescaleMonths)
     elif isinstance(series, Series):
-        series = series.dropna().copy()  # Create a copy to ensure series is untouched.
+        # Create a copy to ensure series is untouched.
+        # Set dtype to avoid conflict when assinging SGI values
+        series = series.copy().dropna(inplace=True).astype(float)
```

Comment:
remove `inplace=True`, otherwise the `dropna` method returns `None` which returns an `AttributeError` if the `astype` method is called